src/parser/ugrammar.y

```diff
@@ -1510,40 +1510,40 @@ args.opt:
 | List of identifiers.  |
 `----------------------*/
 
-%type <::ast::symbols_type> identifiers;
+%type <ast::symbols_type> identifiers;
 identifiers:
   /* empty */ { /* empty */ }
 | identifiers "identifier" { std::swap($$, $1); $$.push_back($2); }
 ;
 
-%type <::ast::rExp> typespec;
+%type <ast::rExp> typespec;
 typespec:
   ":" exp { $$ = $2;}
 ;
 
-%type <::ast::rExp> typespec.opt;
+%type <ast::rExp> typespec.opt;
 typespec.opt:
   /* empty */ { $$=0;}
 | typespec { std::swap($$, $1);}
 ;
 
-%type <::ast::Formal> formal;
+%type <ast::Formal> formal;
 formal:
- var.opt "identifier" typespec.opt  { $$ = ::ast::Formal($2, 0, $3);  }
-| var.opt "identifier" "=" exp typespec.opt  { $$ = ::ast::Formal($2, $4, $5); }
-| var.opt "identifier" "[" "]"  { $$ = ::ast::Formal($2, true); }
+ var.opt "identifier" typespec.opt  { $$ = ast::Formal($2, 0, $3);  }
+| var.opt "identifier" "=" exp typespec.opt  { $$ = ast::Formal($2, $4, $5); }
+| var.opt "identifier" "[" "]"  { $$ = ast::Formal($2, true); }
 ;
 
 // One or several comma-separated identifiers.
-%type <::ast::Formals*> formals.0 formals.1 formals;
+%type <ast::Formals*> formals.0 formals.1 formals;
 formals.1:
-  formal                 { $$ = new ::ast::Formals(1, $1); }
+  formal                 { $$ = new ast::Formals(1, $1); }
 | formals.1 "," formal   { std::swap($$, $1); *$$ << $3; }
 ;
 
 // Zero or several comma-separated identifiers.
 formals.0:
-  /* empty */          { $$ = new ::ast::Formals; }
+  /* empty */          { $$ = new ast::Formals; }
```

Comment:
At some point, it would be nice to convert these `/* empty. */` into proper `%empty`.